editoast/src/views/train_schedule.rs

```diff
@@ -428,6 +403,47 @@ pub async fn train_simulation_batch(
     )
     .await?;
 
+    get_train_batch_simulation(
+        &mut valkey_conn,
+        core.clone(),
+        infra,
+        train_schedules,
+        rolling_stocks,
+        pathfinding_results,
+        electrical_profile_set_id,
+        Some(physics_consist_parameters),
+    )
+    .await
+}
+
+async fn retrieve_rolling_stocks_from_train_schedules(
+    conn: &mut DbConnection,
+    train_schedules: &[TrainSchedule],
+) -> Result<HashMap<String, RollingStockModel>> {
+    let rolling_stocks_ids = train_schedules
+        .iter()
+        .map::<String, _>(|t| t.rolling_stock_name.clone());
+    let (rolling_stocks, _): (Vec<_>, _) =
+        RollingStockModel::retrieve_batch(conn, rolling_stocks_ids).await?;
+
+    let rolling_stocks: HashMap<_, _> = rolling_stocks
+        .into_iter()
+        .map(|rs| (rs.name.clone(), rs))
+        .collect();
+
+    Ok(rolling_stocks)
+}
+
+async fn get_train_batch_simulation(
+    valkey_conn: &mut ValkeyConnection,
+    core: Arc<CoreClient>,
+    infra: &Infra,
+    train_schedules: &[TrainSchedule],
+    rolling_stocks: HashMap<String, RollingStockModel>,
+    pathfinding_results: Vec<PathfindingResult>,
+    electrical_profile_set_id: Option<i64>,
+    physics_consist_parameters: Option<PhysicsConsistParameters>,
```

Comment:
```suggestion
    train_schedules: &[TrainSchedule],
    pathfinding_results: Vec<PathfindingResult>,
    electrical_profile_set_id: Option<i64>,
```

The `get_train_batch_simulation` should simply build the PhysicsConsistParameters from the rolling stock names then call the `consist_train_simulation_batch`